external/EffectiveSource/kerr-circular.c

```diff
@@ -0,0 +1,848 @@
+/*******************************************************************************
+ * Copyright (C) 2012 Barry Wardell
+ ******************************************************************************/
+
+#include <math.h>
+#include <assert.h>
+#include "effsource.h"
+#include <stdio.h>
+#include <gsl/gsl_sf_ellint.h>
+
+/* The particle's coordinate location and 4-velocity */
+static struct coordinate xp;
```

Comment:
Just an FYI, I'm worried all of this code isn't threadsafe., but I haven't looked in detail and I don't know if it matters :)

Reply:
yep probably not threadsafe, I've been working around that when calling the functions